### PR TITLE
updating development document for clarity and legibility

### PR DESCRIPTION
**Pull request by slickwarren**
Updating the development docs with:
* organization based on OS
* adding helpful links for linux based installs

### CODE REVIEW

**Comment by slickwarren** [comment]
looks like there's some similar work done here: https://github.com/jmorganca/ollama/pull/1506 ; these PRs seem to support eachother.

File: docs/development.md [hunk 1]
```diff
@@ -1,22 +1,44 @@
...
 ```bash
 brew install go cmake gcc
 ```
 
+## Linux Prerequisites
+
+- Install cmake and gcc
+  - run `sudo apt-get install cmake gcc`
+- Install [golang](https://go.dev/doc/install)
+- Optionally, Install nvidia-cuda-toolkit for gpu support
```
Comment (antonkesy) [inline docs/development.md:26]: nitpick: Why is "install" upper-case

Cases are not quite uniform 

also:
GPU was UpperCase before changes

File: docs/development.md [hunk 2]
```diff
@@ -1,22 +1,44 @@
...
+## Mac Prerequisites
+
 ```bash
 brew install go cmake gcc
 ```
 
+## Linux Prerequisites
+
+- Install cmake and gcc
+  - run `sudo apt-get install cmake gcc`
```
Comment (antonkesy) [inline docs/development.md:24]: apt is not used by all Linux distributions

maybe: either add alternatives, remove specific install commands, or specify version?

File: docs/development.md [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
+- Install [golang](https://go.dev/doc/install)
+- Optionally, Install nvidia-cuda-toolkit for gpu support
+  - check [system requirements](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements)
+  - select your system and follow [package manager install steps](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#package-manager-installation)
+
+
+## Local Development
+
+- Fork this repo using github's UI
```
Comment (antonkesy) [inline docs/development.md:33]: Why fork it?

Comment (slickwarren) [reply docs/development.md:33]: its a pretty common practice for open source repos. From what I've seen, this helps community users know which branches are 'safe', and also helps keep the branches clean in the original repo.

File: docs/development.md (hunk 1 above)
Comment (slickwarren) [reply docs/development.md:26]: adjusted case (upper for first character and special cases i.e. `GPU`)

File: docs/development.md (hunk 3 above)
Comment (antonkesy) [reply docs/development.md:35]: Yeah, agree
but I think it's restricted who can push (new) branches anyway
so new contributors without any special privileges *have* to create their own fork

I don't want to start a disputation about details; I got my answer and it's a valid reason

**Review by antonkesy — APPROVED** [review]
LGTM

**Comment by slickwarren** [comment]
rebased + updated to reflect latest changes to the readme.

**Pull request closed without merging**